### DISCUSSION TITLE
Could we have a general footer field?

### DISCUSSION THREAD

**Discussion by twMat** [TiddlyWiki/TiddlyWiki5#6965] [Ideas]
I *often* create special conditional ViewTemplates and quite often I want these at bottom of tiddler. I believe this is pretty common (e.g for backlinks). To my surprise I just discovered the **footer** field, [described like](https://tiddlywiki.com/prerelease/#TiddlerFields)

`footer | The footer text for a wizard`

**Question/Idea**
Could this field be repurposed to function as a built-in footer ViewTemplate for any (current) tiddler? I.e if the tiddler has content in this field, it is shown in view mode, just like the text field, but at bottom.

BTW, what distinguishes *wizards* from other tids? What triggers the footer field there?

Alternatively, how about the idea to introduce a footer field (but possibly named sth else)? I believe the popularity warrants a standard field.

**Comment by Jermolene** [comment]
Hi @twMat "wizard" was an early name for what we now call "modals"; we should revise any mentions now to use "modals".

> Could this field be repurposed to function as a built-in footer ViewTemplate for any (current) tiddler? I.e if the tiddler has content in this field, it is shown in view mode, just like the text field, but at bottom.

I am not sure that this would be very useful. It only accommodates a very narrow set of use cases; basically, where the footer is known in its entirety at the time of writing the tiddler. I think it's much more common to want to add a dynamic footer which uses the same template for each tiddler.

**Reply by twMat** [reply]
Thanks @Jermolene . The mention of "wizard" is from $:/language/Docs/Fields/footer (not a simple doc to edit) - but searching for the term reveals a few instances, including the tiddlers SampeWizard and SampleWizard2

> I think it's much more common to want to add a dynamic footer which uses the same template for each tiddler.

You mean a regular ViewTemplate, or what else? One thing I hope we can avoid is to force users to create "conditional" viewtemplates. A regular user should not have to learn this trick to create a footer.

Could the user have the best of two worlds? Comparable to how *lists* can be "locally" overridden by using the `list-before` etc, could something be made so that **if** the user defines a "local" footer field, then this takes precedence over the default footer template you refer to? I guess this moves the conditional test to system level by using the template as a fallback.

**Reply by Jermolene** [reply]
> Thanks @Jermolene . The mention of "wizard" is from $:/language/Docs/Fields/footer (not a simple doc to edit) - but searching for the term reveals a few instances, including the tiddlers SampeWizard and SampleWizard2

Thanks @twMat I've updated the field description docs here:  https://github.com/Jermolene/TiddlyWiki5/commit/4007610d522281204bdef90e11b68834cfee92b7

I don't think we need to change "SampleWizard" etc. because the docs use "modal (or wizard)" fairly consistently

> You mean a regular ViewTemplate, or what else? One thing I hope we can avoid is to force users to create "conditional" viewtemplates. A regular user should not have to learn this trick to create a footer.

I'm not sure I agree. Because it's static, the "footer" field you propose here is not much more functional than just moving the cursor to the bottom of the text field and typing the footer there. Teaching how to make a view template segment opens up much more powerful and useful capabilities.

> Could the user have the best of two worlds? Comparable to how _lists_ can be "locally" overridden by using the `list-before` etc, could something be made so that **if** the user defines a "local" footer field, then this takes precedence over the default footer template you refer to? I guess this moves the conditional test to system level by using the template as a fallback.

The "default footer template" is just a view template segment that has been added to the core. The core cannot extend it because it's a hypothetical thing added by a user.
